rna_cd/bam_process.py

```diff
@@ -64,37 +64,42 @@ def softclip_bases(reader: AlignmentFile, contig: str,
 def coverage(reader: AlignmentFile, contig: str, region: Tuple[int, int],
              method: Callable = np.mean) -> int:
     """Calculate average/median/etc coverage for a region"""
-    covs = reader.count_coverage(contig=contig, start=region[0],
-                                 stop=region[1])
+    start, end = region
+    covs = reader.count_coverage(contig=contig, start=start, stop=end)
 
     return method(np.sum(covs))
 
 
 def process_bam(path: Path, chunksize: int = 100,
                 contig: str = "chrM") -> np.ndarray:
-    """Process bam file to an ndarray"""
+    """
+    Process bam file to an ndarray
+
+    :returns: numpy ndarray of shape (n_features,)
+    """
     echo("Calculating features for {0}".format(path.name))
     reader = AlignmentFile(str(path))
     try:
-        ctg_idx = reader.references.index(contig)
+        contig_idx = reader.references.index(contig)
     except ValueError:
         raise ValueError("Contig {0} does not exist in BAM file".format(
             contig
         ))
-    contig_size = reader.lengths[ctg_idx]
+    contig_size = reader.lengths[contig_idx]
 
-    arr = []
+    full_array = []
```

Comment:
unless `full_array` is some sort of machine learning terminology, this name is not very informative. Especially as this code snippet translates to `full_array = empty list`. Python lists do not have a pre-determined length, so can never be full anyway.